docs/source/API/containers/Array.rst

```diff
@@ -0,0 +1,173 @@
+..
+  Use the following convention for headings:
+
+    # with overline, for parts (collections of chapters)
+
+    * with overline, for chapters
+
+    = for sections
+
+    - for subsections
+
+    ^ for subsubsections
+
+    " for paragraphs
+
+..
+  Class / method / container name)
+  for free functions that are callable, preserve the naming convention, `view_alloc()`
+
+``Array``
+==============
+
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
+..
+  The (pulic header) file the user will include in their code
+
+Header File: ``Kokkos_Core.hpp``
+
+..
+  High-level, human-language summary of what the thing does, and if possible, brief statement about why it exists (2 - 3 sentences, max);
+
+Description
+-----------
+
+``Array`` is a contiguous aggregate owning container storing a fixed size sequence of objects (models holding exactly N elements).
+
+* This container is an owning container (the data is embeddded in the container itself).
+* This container is an aggregate type with the same semantics as a struct holding a C-style array ``T[N]`` as its only non-static data member when ``N > 0``; otherwise, it is an empty container.
+* Unlike a C-style array, it doesn't decay to ``T*`` automatically.
+* As an aggregate type, it can be initialized with aggregate-initialization given at most ``N`` initializers that are convertible to ``T``: ``Kokkos::Array<int, 3> a = { 1, 2, 3 };``.
+
+..
+  The API of the entity.
+
+Interface
+---------
+
+..
+  The declaration or signature of the entity.
+
+.. cppkokkos:struct:: template <class T, size_t N> Array
+
+  ..
+    Template parameters (if applicable)
+    Omit template parameters that are just used for specialization/are deduced/ and/or should not be exposed to the user.
+
+  .. rubric:: Template Parameters
+
+  :tparam T: The type of the element being stored.
+  :tparam N: The number of elements being stored.
+
+  .. rubric:: Public Types
+
+  .. cppkokkos:type:: value_type = T
+  .. cppkokkos:type:: pointer = T*
+  .. cppkokkos:type:: const_pointer = const T*
+  .. cppkokkos:type:: reference = T&
+  .. cppkokkos:type:: const_reference = const T&
+  .. cppkokkos:type:: size_type = size_t
+  .. cppkokkos:type:: difference_type = ptrdiff_t
+
+  .. rubric:: Public Member Functions
+
+  .. cppkokkos:function:: static constexpr bool empty()
+
+    :return: ``N == 0``
```

Comment:
this always returns `false`, right?

Reply:
Passing 0 as the template parameter is perfectly valid, so N == 0 really is the return value (unless I got it wrong, like our implementation of Array had it for years...)

Reply:
https://github.com/kokkos/kokkos/blob/0766bc570e25b7d952d747fffd8c0828b4dcaf34/core/src/Kokkos_Array.hpp#L107
AFAIU we hardcoded `false`

Reply:
There is a specialization for 0-sized arrays which does properly return `true`.
https://github.com/kokkos/kokkos/blob/0766bc570e25b7d952d747fffd8c0828b4dcaf34/core/src/Kokkos_Array.hpp#L175
Apparently, it is not covered in our test though.